src/olympia/addons/tasks.py

```diff
@@ -425,3 +428,85 @@ def add_firefox57_tag(ids, **kw):
+            # find...
+            # Note that we can't use parse_addon() wrapper because it no longer
+            # exposes the real value of `strictCompatibility`...
+            path = addon.current_version.all_files[0].file_path
+            with storage.open(path) as file_:
+                extracted_dir = extract_zip(file_)
+            parser = RDFExtractor(extracted_dir)
+            if parser.find('strictCompatibility') != 'true':
+                # It had not enabled strict compatibility. That means we should
+                # bump it!
+                application_versions.max = new_max_appver
+                application_versions.save()
+                did_update = True
+            else:
+                # Strict compatibility was enabled for this add-on, let's avoid
+                # bumping, and note that it did not need updating.
+                did_update = False
+            rm_local_tmp_dir(extracted_dir)
+        except Exception as exp:
+            # A number of things can go wrong missing file, path somehow not
+            # existing, etc). In any case, that means the add-on is in a weird
+            # state and should be ignored (this is a one off task).
+            log.error(u'bump_appver_for_legacy_addons: ignoring addon %d, '
+                      u'received %s when bumping.', addon.pk, unicode(exp))
+    return did_update
+
+
+# Rate limit is per-worker. Kept low to not overload the database with updates.
+# We have 5 workers in the default queue, we have roughly 25.000 add-ons to go
+# through, since process_addons() chunks contain 100 add-ons the task should be
+# fired 250 times. With 5 workers at 5 tasks / minute limit we should do 25
+# tasks in a minute, taking ~ 10 minutes for the whole thing to finish.
+@task(rate_limit='5/m')
+@write
+def bump_appver_for_legacy_addons(ids, **kw):
+    """
+    Task to bump the max appversion to 56.* for legacy add-ons that have not
+    enabled strictCompatibility in their manifest.
+    """
+    addons = Addon.objects.filter(id__in=ids)
+    # The AppVersions we want to point to now.
+    new_max_appversions = {
+        amo.FIREFOX.id: AppVersion.objects.get(
+            application=amo.FIREFOX.id, version='56.*'),
+        amo.ANDROID.id: AppVersion.objects.get(
+            application=amo.ANDROID.id, version='56.*')
+    }
+
+    addons_to_reindex = []
+    for addon in addons:
+        did_update_for_android = None
+        did_update_for_firefox = bump_appver_for_addon_if_necessary(
+            addon, amo.FIREFOX.id, new_max_appversions[amo.FIREFOX.id])
+        # If did_update_for_firefox is False (and not just None), that means we
+        # looked at the package and figured it had strictlyCompatible enabled,
+        # so we don't need to do any bumping, there is nothing else to do.
+        # Otherwise we continue with Android.
+        if did_update_for_firefox is not False:
+            did_update_for_android = bump_appver_for_addon_if_necessary(
```

Comment:
I realise Android is an edge case, but why do we need to extract the RDF, etc, a second time if we know it needs bumping?

Reply:
Also edge case within an edge case: `ANDROID` has different compatibility range than `FIREFOX`.  How is that handled?

Reply:
Regarding extracting twice: yeah, that's unfortunate. I'll see if I can avoid that.

Different compatibility ranges should be handled, since we check compat for each app and bump separately.